### PR TITLE
Add geckodriver, related config, and flags.

### PR DESCRIPTION
**Pull request by mgiambalvo**
geckodriver is still a little experimental, so I've disabled it by default. You need "marionette: true" in your capabilities in order to use it. Also, there's a bug where after we delete the session at the end of the test, geckodriver won't run again so you need to restart selenium.

### CODE REVIEW

File: lib/cmds/opts.ts [hunk 1]
```diff
@@ -47,16 +48,19 @@ opts[STANDALONE] = new Option(
     STANDALONE, 'Install or update selenium standalone', 'boolean', StandAlone.isDefault);
 opts[CHROME] =
     new Option(CHROME, 'Install or update chromedriver', 'boolean', ChromeDriver.isDefault);
+opts[GECKO] = new Option(GECKO, 'Install or update geckodriver', 'string', GeckoDriver.isDefault);
```
Comment (cnishina) [inline lib/cmds/opts.ts]: the 3rd argument should then be 'boolean'

File: lib/cmds/update.ts [hunk 2]
```diff
@@ -31,6 +31,10 @@ let prog = new Program()
                .addOption(Opts[Opt.ANDROID_ABIS])
                .addOption(Opts[Opt.ANDROID_ACCEPT_LICENSES]);
 
+if (GeckoDriver.supports(os.type(), os.arch())) {
+  prog.addOption(Opts[Opt.VERSIONS_GECKO]).addOption(Opts[Opt.GECKO])
```
Comment (cnishina) [inline lib/cmds/update.ts]: semi-colon... running gulp should fix this.

File: lib/cmds/opts.ts (hunk 1 above)
Comment (mgiambalvo) [inline lib/cmds/opts.ts]: done

File: lib/cmds/update.ts (hunk 2 above)
Comment (mgiambalvo) [inline lib/cmds/update.ts]: done

File: lib/cmds/start.ts [hunk 3]
```diff
@@ -133,6 +133,11 @@ function start(options: Options) {
       args.push('-Dwebdriver.chrome.logfile=' + chromeLogs);
     }
   }
+  if (downloadedBinaries[GeckoDriver.id] != null) {
+    args.push(
+        '-Dwebdriver.gecko.driver=' +
```
Comment (cnishina) [inline lib/cmds/start.ts]: Thanks for fixing this!

**Comment by cnishina** [comment]
Please fix the commit message and fix conflicts.

**Comment by mgiambalvo** [comment]
Note that this fixes #52

**Pull request merged**